src/casp/services/utils/email_utils.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import typing as t
+
+from sendgrid import SendGridAPIClient
+from sendgrid.helpers.mail import Mail
+
+logger = logging.getLogger(__name__)
+
+TEMPLATE_DIR = "src/casp/services/admin/templates"
+
+
+class EmailSender:
+
+    def __init__(
+        self,
+        sendgrid_client: t.Optional[t.Union[SendGridAPIClient, str]],
```

Comment:
Why support the option of an API key or a client instead of just restricting it to one of them?

Reply:
That makes sense.  In that case, would it make more sense to restrict it to the API key?

Reply:
Or could we import settings in this module to get the API key that way?

Reply:
I guess if the thinking is we want to avoid importing the Sendgrid client outside this module, what's the benefit to allowing it as an input?  Is there a reason we might want to use the same Sendgrid client across modules?

Reply:
I actually started with it as a string but the problem is for testing, I wanted to be able to pass in a mocked SendGrid client.  Maybe if I add a method getting the client and mock that it would be better

Reply:
Actually, came up with a better way. I can just mock the `sendgrid.SendGridAPIClient` constructor from the test (just have to make sure to change how I do imports a little but it seems to work)

Reply:
Ooh that makes a lot of sense